src/tqec/compile/tree/tree.py

```diff
@@ -257,7 +258,7 @@ def _generate_annotations(
         only_use_database: bool = False,
         lookback: int = 2,
         add_polygons: bool = False,
-        parallel: bool = False,
+        parallel_process_count: int = 1,
```

Comment:
Either:

- add an explicit note (or even warning) in the docstring saying that this setting is automatically reset to 1 when a detector database is provided or,
- "raise" a warning when the database is provided and this is not `1`.

I personally prefer option 1

Reply:
Ho right, I missed the fact that this was a private function. Then disregard my comment, the current status is perfectly fine for me, sorry for missing that detail.

I'll try to find time to test the PR from a user perspective today, and if everything is alright I'll approve it.